sdk/containerregistry/container-registry/src/blob/containerRegistryBlobClient.ts

```diff
@@ -0,0 +1,400 @@
+      ...options,
+      loggingOptions: {
+        logger: logger.info,
+        // This array contains header names we want to log that are not already
+        // included as safe. Unknown/unsafe headers are logged as "<REDACTED>".
+        additionalAllowedQueryParameters: [
+          "last",
+          "n",
+          "orderby",
+          "digest",
+          "_nouploadcache",
+          "_state",
+        ],
+      },
+    };
+    // Require audience now until we have a default ACR audience from the service.
+    if (!options.audience) {
+      throw new Error(
+        "ContainerRegistryClientOptions.audience must be set to initialize ContainerRegistryClient."
+      );
+    }
+
+    const defaultScope = `${options.audience}/.default`;
+    const serviceVersion = options.serviceVersion ?? LATEST_API_VERSION;
+    const authClient = new GeneratedClient(endpoint, serviceVersion, internalPipelineOptions);
+    this.client = new GeneratedClient(endpoint, serviceVersion, internalPipelineOptions);
+    this.client.pipeline.addPolicy(
+      bearerTokenAuthenticationPolicy({
+        credential,
+        scopes: [defaultScope],
+        challengeCallbacks: new ChallengeHandler(
+          new ContainerRegistryRefreshTokenCredential(authClient, defaultScope, credential)
+        ),
+      })
+    );
+  }
+
+  /**
+   * Delete a blob.
+   * @param digest - the digest of the blob to delete
+   * @param options - optional configuration used to send requests to the service
+   */
+  public async deleteBlob(digest: string, options?: DeleteBlobOptions): Promise<void> {
+    const { span, updatedOptions } = createSpan("ContainerRegistryBlobClient-deleteBlob", options);
+
+    try {
+      await this.client.containerRegistryBlob.deleteBlob(
+        this.repositoryName,
+        digest,
+        updatedOptions
+      );
+    } catch (e) {
+      span.setStatus({ code: SpanStatusCode.ERROR, message: e.message });
+    } finally {
+      span.end();
+    }
+  }
+
+  /**
+   * Upload a manifest for an OCI artifact.
+   *
+   * @param manifest - the manifest to upload. If a resettable stream (a factory function that returns a stream) is provided, it may be called multiple times. Each time the function is called, a fresh stream should be returned.
+   */
+  public async uploadManifest(
+    manifest: (() => NodeJS.ReadableStream) | NodeJS.ReadableStream | OciManifest,
+    options?: UploadManifestOptions
+  ): Promise<UploadManifestResult> {
+    const { span, updatedOptions } = createSpan(
+      "ContainerRegistryBlobClient-uploadManifest",
+      options
+    );
+
+    try {
+      let manifestBody: Buffer | NodeJS.ReadableStream;
+
+      if (isReadableStream(manifest)) {
+        manifestBody = await readStreamToEnd(manifest);
+      } else if (typeof manifest === "function") {
+        manifestBody = await readStreamToEnd(manifest());
+      } else {
+        const serialized = serializer.serialize(Mappers.OCIManifest, manifest);
```

Comment:
interesting, I don't think we've had to do this in convenience before. I wonder if this is a pattern we should think about how to generalize more.